test/integration/test_user_defined_output_plugins.py

```diff
@@ -0,0 +1,185 @@
+      "--termination-predicate", "benchmark.http_2xx:24", "--rps", "100",
+      "--user-defined-plugin-config", fake_plugin_config, "--user-defined-plugin-config",
+      logging_plugin_config
+  ])
+  counters = http_test_server_fixture.getNighthawkCounterMapFromJson(parsed_json)
+  asserts.assertCounterGreaterEqual(counters, "benchmark.http_2xx", 25)
+
+  user_defined_outputs_by_result = getUserDefinedOutputsFromJson(parsed_json)
+  asserts.assertEqual(len(user_defined_outputs_by_result), 1)
+
+  global_user_defined_outputs = user_defined_outputs_by_result['global']
+  asserts.assertEqual(len(global_user_defined_outputs), 2)
+  for user_defined_output in global_user_defined_outputs:
+    if user_defined_output["plugin_name"] == "nighthawk.fake_user_defined_output":
+      asserts.assertGreaterEqual(user_defined_output["typed_output"]["data_called"], 25)
+      asserts.assertGreaterEqual(user_defined_output["typed_output"]["headers_called"], 25)
+    else:
+      # There should not be any plugins that don't have the two above names.
+      asserts.assertEqual(user_defined_output["plugin_name"],
+                          "nighthawk.log_response_headers_plugin")
+
+
+@pytest.mark.parametrize('server_config',
+                         ["nighthawk/test/integration/configurations/nighthawk_http_origin.yaml"])
+def test_handle_headers_failure_increments_counter(http_test_server_fixture):
+  """Checks that counters are incremented properly for handleResponseHeaders failures."""
+  fake_plugin_config = (
+      "{name:\"nighthawk.fake_user_defined_output\","
+      "typed_config:{\"@type\":\"type.googleapis.com/nighthawk.FakeUserDefinedOutputConfig\","
+      "fail_headers:true,header_failure_countdown:9}}")
+
+  parsed_json, _ = http_test_server_fixture.runNighthawkClient([
+      http_test_server_fixture.getTestServerRootUri(), "--duration", "100", "--concurrency", "1",
+      "--termination-predicate", "benchmark.user_defined_plugin_handle_headers_failure:10", "--rps",
+      "100", "--user-defined-plugin-config", fake_plugin_config
+  ])
+  counters = http_test_server_fixture.getNighthawkCounterMapFromJson(parsed_json)
+  asserts.assertCounterGreaterEqual(counters, "benchmark.http_2xx", 20)
+
+  user_defined_outputs_by_result = getUserDefinedOutputsFromJson(parsed_json)
+  asserts.assertEqual(len(user_defined_outputs_by_result), 1)
+
+  global_user_defined_outputs = user_defined_outputs_by_result["global"]
+  asserts.assertEqual(len(global_user_defined_outputs), 1)
+  user_defined_output = global_user_defined_outputs[0]
+  asserts.assertEqual(user_defined_output["plugin_name"], "nighthawk.fake_user_defined_output")
+  asserts.assertGreaterEqual(user_defined_output["typed_output"]["data_called"], 20)
+  asserts.assertGreaterEqual(user_defined_output["typed_output"]["headers_called"], 20)
+
+  expected_failure_counter = user_defined_output["typed_output"]["headers_called"] - 9
```

Comment:
Optional:
Should this operation for calculating the expected value have a comment? Deducting the value 9 feels a bit magical for someone who is unfamiliar with the codebase.